rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -416,7 +436,7 @@ private void execError(String err) {
 
     private void go() {
         // TODO: transaction call for pre-compiled  contracts
-        if (vm == null) {
+        if (vm == null || program.getResult().getException() != null) {
```

Comment:
why do we need this?

Reply:
Ops, we don't need it anymore. It's a leftover from the several changes made. Good catch, thanks, I will fix it.